.github/workflows/ci.yaml

```diff
@@ -172,25 +178,34 @@ jobs:
 
       - name: Get version from pyproject.toml for main
         id: main_version
-        run: echo "VERSION=$(poetry version -s)" >> $GITHUB_OUTPUT
+        run: |
+          VERSION=$(poetry version -s)
+          echo "VERSION=$VERSION" >> $GITHUB_OUTPUT
+          echo "Main version is $VERSION"
 
       - name: Get is_version_bumped
         id: is_version_bumped
         run: |
           if [ "${{ steps.pr_version.outputs.VERSION }}" == "${{ steps.main_version.outputs.VERSION }}" ]; then
-            echo "IS_VERSION_BUMPED=false" >> $GITHUB_OUTPUT
+            BUMPED=false
           else
-            echo "IS_VERSION_BUMPED=true" >> $GITHUB_OUTPUT
+            BUMPED=true
           fi
+          echo "IS_VERSION_BUMPED=$BUMPED" >> $GITHUB_OUTPUT
+          echo "IS_VERSION_BUMPED is $BUMPED"
 
-      - uses: actions/checkout@v4
+      - name: Recheckout PR branch
+        uses: actions/checkout@v4
 
   bump-pre-release-version:
     permissions:
       contents: write # Allow us to write to the repository
     needs: check-version-is-bumped
     runs-on: ubuntu-latest
-    if: ${{ needs.check-version-is-bumped.outputs.is_version_bumped == 'false' && !contains(github.event.pull_request.labels.*.name, 'no version bump') && !contains(github.event.pull_request.labels.*.name, 'dependencies') && github.event_name != 'schedule' }}
+    # The github action containerize-agent will push to dockerhub on each commit to main. That
+    # means we almost always want to bump the version on a PR since, if the current version is
+    # a release version, we will overwrite that release version with this new one.
+    if: ${{ needs.check-version-is-bumped.outputs.is_version_bumped == 'false' && !contains(github.event.pull_request.labels.*.name, 'no version bump') && github.event_name != 'schedule' }}
```

Comment:
This is the functional change in this file. We remove the condition that `dependencie` PRs shouldn't bump the version number. The reason this can be problematic is explained in the new comment. We basically always want to bump the version on a PR since otherwise we may overwrite the release tag with this new image on dockerhub.